packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts

```diff
@@ -268,10 +270,37 @@ export class AmplifyAuthTransform extends AmplifyCategoryTransform {
     } else if (_.isEmpty(this._cognitoStackProps.triggers)) {
       parameters = Object.assign(parameters, { triggers: JSON.stringify(this._cognitoStackProps.triggers) });
     }
+
+    this.validateCfnParameters(context, oldParameters, parameters);
+
     //save parameters
     JSONUtilities.writeJson(parametersJSONFilePath, parameters);
   };
 
+  public validateCfnParameters(context: $TSContext, oldParameters: $TSAny, parametersJson: $TSAny) {
+    // There was a bug between v7.3.0 and v7.6.9 where Cognito resources were being created with incorrect `requiredAttributes` parameter
+    // Since `requiredAttributes` is immutable, we must adjust this or CloudFormation step will fail
+    // More info: https://github.com/aws-amplify/amplify-cli/issues/9525
+    if (!oldParameters?.requiredAttributes?.length) {
+      return true;
```

Comment:
I think you can drop the return type completely here since the function will either succeed or terminate the process.